File: Backend/Data.MyShop/Entities/ProductEntity.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Data.MyShop.Entities
{
    [Table("tbl_Products")]
    public class ProductEntity : BaseEntity<int>
    {
        public string Description { get; set; }
        public double Price { get; set; }

        [DisplayName("Category")]
        //Every product have a category
        public CategoryEntity Category { get; set; }

        [ForeignKey(nameof(Category))]
        public int? CategoryId { get; set; }
        public virtual ICollection<ProductImageEntity> ProductImages { get; set; }
        public virtual ICollection<CommentEntity> Comments { get; set; }
    }
}

File: Backend-Java/src/main/java/org/example/DAL/entities/ProductEntity.java
package org.example.DAL.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data //lombok {get; set;}
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "tbl_Products")
public class ProductEntity extends BaseEntity {
    private String description;
    private double price;
    @ManyToOne
    @JoinColumn(name="category_id", nullable = false)
    private CategoryEntity category;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<ProductImageEntity> productImages;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<CommentEntity> comments;
}
